### DISCUSSION TITLE
Remove Physical and add Physical question

### DISCUSSION THREAD

**Discussion by neciszhang** [pmndrs/use-cannon#247] [Q&A] [answered]
Hi, 
      I have some question with react-cannon. 
      1. I only want to remove a mesh's physical, what should i do? 
      2. Then when i remove the physical, i want to add the mesh physical, what should i do?
      I know when using cannon.js ,i can remove body in physical world ,and reset addBody in physical world.
      Could you help me?Thank you very much!

**Comment by drcmda** [comment, marked as answer]
hooks cannot be conditional in react, normally you put an object "to sleep" if its not supposed to receive physics input, and you wake it up once it has to respond again. i think this would be more efficient than removing it from physics.

otherwise you can unmount (not render) the object and it will be gone. and then remount?

**Reply by neciszhang** [reply]
Thank you very much! It's easy to control.
